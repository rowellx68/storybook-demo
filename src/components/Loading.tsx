/** @jsx jsx */
import React from 'react';
import { css, jsx, keyframes } from '@emotion/core';

const rotate = keyframes({
  from: {
    transform: 'rotate(0deg)'
  },
  to: {
    transform: 'rotate(360deg)'
  }
});

const loading = css({
  padding: '0.5rem',
  display: 'inline-block',
  i: {
    animation: `${rotate} 1s ease-out infinite`,
    fontSize: '1.5rem',
    opacity: 0.5
  }
});

interface LoadingProps {
  children?: React.ReactElement;
}

const Loading: React.FunctionComponent<LoadingProps> = ({ children }) => {
  return (
    <div css={loading}>{children ?? <i className='las la-fw la-sync'></i>}</div>
  );
};

export default Loading;
