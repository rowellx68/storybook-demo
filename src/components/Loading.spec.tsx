import React from 'react';
import { mount } from 'enzyme';

import Loading from 'components/Loading';

describe('Loading', () => {
  it('renders correct icon class', () => {
    const wrapper = mount(<Loading />);

    expect(wrapper.exists('.la-sync')).toEqual(true);
  });

  it('renders the correct mark-up', () => {
    const wrapper = mount(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the correct mark-up with custom spinner', () => {
    const wrapper = mount(
      <Loading>
        <i className='las la-fw la-cog'></i>
      </Loading>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
