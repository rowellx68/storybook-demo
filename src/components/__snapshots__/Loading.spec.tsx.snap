// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders the correct mark-up 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  padding: 0.5rem;
  display: inline-block;
}

.emotion-0 i {
  -webkit-animation: animation-0 1s ease-out infinite;
  animation: animation-0 1s ease-out infinite;
  font-size: 1.5rem;
  opacity: 0.5;
}

<Loading>
  <div
    className="emotion-0"
  >
    <i
      className="las la-fw la-sync"
    />
  </div>
</Loading>
`;

exports[`Loading renders the correct mark-up with custom spinner 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  padding: 0.5rem;
  display: inline-block;
}

.emotion-0 i {
  -webkit-animation: animation-0 1s ease-out infinite;
  animation: animation-0 1s ease-out infinite;
  font-size: 1.5rem;
  opacity: 0.5;
}

<Loading>
  <div
    className="emotion-0"
  >
    <i
      className="las la-fw la-cog"
    />
  </div>
</Loading>
`;
